crates/red_knot_python_semantic/resources/mdtest/annotations/starred_expressions.md

```diff
@@ -0,0 +1,18 @@
+# Starred expression annotations
+
+Type annotations for `*args` can be starred expressions themselves:
+
+```py
+from typing import TypeVarTuple
+
+Ts = TypeVarTuple("Ts")
+
+def append_int(*args: *Ts) -> tuple[*Ts, int]:
```

Comment:
As far as I understand, vararg annotations are the only place where starred expressions can appear in annotations at the "outermost" level (we can have things like `tuple[*Ts]`, but not a "naked" `*Ts`).

Reply:
Yes, I _think_ that's correct!

Reply:
I also think it's correct, but (as I mentioned in Discord) we can't restrict our test cases to valid syntax, because we also need to support error recovery. So when we are adding tests for something that currently causes us to panic, it's fine (even good!) to write a test that is semantically nonsense, or even emits syntax errors, but demonstrates that we don't panic.

It would be helpful if mdtest gained the feature to assert on syntax errors, instead of just immediately failing if there are any.

(To be clear, I think this test is a good addition regardless, and I don't even feel strongly about adding the invalid-syntax test in this PR. Just observing that it is possible, and useful, to do so.)